test/Core.Test/Services/GroupServiceTests.cs

```diff
@@ -30,12 +33,116 @@ public GroupServiceTests()
             );
         }
 
-        // Remove this test when we add actual tests. It only proves that
-        // we've properly constructed the system under test.
         [Fact]
-        public void ServiceExists()
+        public async Task SaveAsync_DefaultGroupId_CreatesGroupInRepository()
         {
-            Assert.NotNull(_sut);
+            var organization = new Models.Table.Organization
+            {
+                Id = Guid.NewGuid(),
+                UseGroups = true,
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            await _sut.SaveAsync(group);
+
+            await _groupRepository.Received().CreateAsync(group);
+        }
+
+        [Fact]
+        public async Task SaveAsync_NonExistingOrganizationId_ThrowsBadRequest()
+        {
+            var group = new Models.Table.Group
+            {
+                Id = Guid.NewGuid(),
+            };
+
+            await Assert.ThrowsAsync<BadRequestException>(() => _sut.SaveAsync(group));
```

Comment:
For these failure cases, it would be great to validate that no event or repository work is done. NSubstitute seems to lack a strict mode or `VerifyNoOtherCalls` assertion, so I think the best way is to use a `DidNotReceiveWithAnyArgs` stack:
```
await _groupRepository.DidNotReceiveWithAnyArgs().CreateAsync(default);
await _groupRepository.DidNotReceiveWithAnyArgs().ReplaceAsync(default);
await _eventService.DidNotReceiveWithAnyArgs().LogGroupEventAsync(default, default, default);
```

As a side note, this is one reason AutoFixture and the SutProvider are cool -- you don't need to worry about resetting your substitutions since they're fresh every time.